tools/ideabuck/src/com/facebook/buck/intellij/ideabuck/run/BuckRunConfigurationExtension.java

```diff
@@ -0,0 +1,207 @@
+        getBuckQueryCommandHandler(currentProject, buckFile);
+    buckQueryCommandHandler.command().addParameter("owner(" + containingFile.getPath() + ")");
+    buildManager.runInCurrentThreadPostEnd(
+        buckQueryCommandHandler,
+        () -> {
+          if (targetResults != null) {
+            for (String target : targetResults) {
+              BuckModule buckModule = currentProject.getComponent(BuckModule.class);
+              buckModule.attach(target);
+
+              ResultCallbackBuckHandler buckAuditCommandHandler =
+                  getBuckAuditCommandHandler(currentProject, buckFile, additionalClasspath, target);
+              buckAuditCommandHandler.command().addParameter("classpath");
+              buckAuditCommandHandler.command().addParameter(target);
+              buckAuditCommandHandler.command().addParameter("--json");
+              buildManager.runInCurrentThreadPostEnd(buckAuditCommandHandler,
+                  () -> {
+                  });
+            }
+          }
+        });
+    appendClasspath(javaParameters, currentProject, additionalClasspath);
+  }
+
+  private void appendClasspath(JavaParameters javaParameters, Project currentProject,
+      List<String> additionalClasspath) {
+    final PathsList classpath = javaParameters.getClassPath();
+    for (String jar : additionalClasspath) {
+      classpath.add(jar);
+    }
+    JsonObject jsonObject = null;
+    try {
+      jsonObject = new JsonParser()
+          .parse(new FileReader(tempFile.getPath()))
+          .getAsJsonObject();
+    } catch (FileNotFoundException e) {
+      return;
+    }
+    for (String target : targetResults) {
+      if (jsonObject == null || jsonObject.getAsJsonObject("results") == null
+          || jsonObject.getAsJsonObject("results").getAsJsonObject(target) == null
+          || jsonObject.getAsJsonObject("results").getAsJsonObject(target).get("output")
+          == null) {
+        return;
+      }
+      classpath.add(currentProject.getBasePath() + "/" + jsonObject.getAsJsonObject("results")
+          .getAsJsonObject(target).get("output").getAsString());
+    }
+  }
+
+  @NotNull
+  private ResultCallbackBuckHandler getBuckAuditCommandHandler(Project currentProject,
+      VirtualFile buckFile, List<String> additionalClasspath, String target) {
+    return new ResultCallbackBuckHandler(
+        currentProject,
+        buckFile.getParent(),
+        BuckCommand.AUDIT,
+        new FutureCallback<String>() {
+          @Override
+          public void onSuccess(@Nullable String result) {
+            if (result == null
+                || result.isEmpty()) {
+              return;
+            }
+            JsonArray classpathArray = new JsonParser().parse(result)
+                .getAsJsonObject().getAsJsonArray(target);
+            if (classpathArray != null) {
+              classpathArray.forEach(
+                  jsonElement ->
+                      additionalClasspath.add(jsonElement.getAsString())
```

Comment:
As per my other comment, I think that some of these elements won't be generated by `--deep`.

If that thinking is correct, then I'm not sure if there are one or two error conditions:

1.  If the dependency is missing, that's because `--deep` only fetched the ABI from the build server.  This seems easy to detect, and could probably be remedied by repeating a more thorough version of the previous build, like `buck build $(buck query 'deps(original-target)')`?
2.  I'm unclear...is it possible for the dependency to exist, but be stale?  This would happen if buck only rebuilt the ABI and not the actual jar.  (I'm not totally sure what buck's guarantee is for `--deep` builds of java_library targets.)